File: app/src/main/java/com/example/getfood/callback/CartItemTouchListener.java
package com.example.getfood.callback;

import com.fazemeright.canteen_app_models.models.CartItem;

public interface CartItemTouchListener {
    void onIncreaseClicked(int adapterPosition);

    void onDecreaseClicked(int adapterPosition);

    void onItemRemoved(int adapterPosition);

    void onItemRemoveUndo(CartItem removedItem, int mRecentlyDeletedItemPosition);
}

File: app/src/main/java/com/example/getfood/callback/CartItemTouchListener.kt
package com.example.getfood.callback

import com.example.canteen_app_models.models.CartItem

open interface CartItemTouchListener {
    fun onIncreaseClicked(adapterPosition: Int)
    fun onDecreaseClicked(adapterPosition: Int)
    fun onItemRemoved(adapterPosition: Int)
    fun onItemRemoveUndo(removedItem: CartItem, mRecentlyDeletedItemPosition: Int)
}